File: src/main/java/gregtechlite/gtlitecore/api/advancement/AdvancementTrigger.java
package gregtechlite.gtlitecore.api.advancement;

import net.minecraft.advancements.ICriterionTrigger;
import net.minecraft.entity.player.EntityPlayerMP;

public interface AdvancementTrigger<T extends AdvancementCriterion> extends ICriterionTrigger<T>
{

    void trigger(EntityPlayerMP player);

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/advancement/AdvancementTrigger.kt
package gregtechlite.gtlitecore.api.advancement

import net.minecraft.advancements.ICriterionTrigger
import net.minecraft.entity.player.EntityPlayerMP

interface AdvancementTrigger<T : AdvancementCriterion> : ICriterionTrigger<T>
{

    fun trigger(player: EntityPlayerMP?)
}